### PR TITLE
Integrate errors view with engine

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description
The IDE now displays errors on nodes basing of the new "executionContext/expressionUpdates".

image

### Important Notes
Added structures for executionContext/executionStatus, so the warnings about unknown notification variant should be no more.
There is known issue: when function graph is edited such way that it results in an error, after going out we don't receive update with "error" payload (however the type is a proper error type).

### Checklist
Please include the following checklist in your PR:

- [ ] The `CHANGELOG.md` was updated with the changes introduced in this PR.
- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [ ] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [ ] ~~All code has been manually tested in the "debug/interface" scene.~~
- [ ] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

**Comment by farmaazon** [comment]
I have one concern with my code so far: once we merge this PR as it is, we won't support the old Engine's version (< 0.2.2) - the projects with that version set won't have even ports colored. The question is, if should I fight for backward compatibility.

**Comment by wdanilo** [comment]
Amazing to see it! Good job to everyone involved in this ❤️ Adam, BIG KUDOS for making it visually correct - that the error descriptions appear only in one place in the graph, that's huge!

> I have one concern with my code so far: once we merge this PR as it is, we won't support the old Engine's version (< 0.2.2) - the projects with that version set won't have even ports colored. The question is, if should I fight for backward compatibility.

No, we don't care about backward compatibility before public release. However, it is important that people who were running Enso so far, after downloading the new version will have everything working (even having local projects from the old version on disk) - if this will not work, this will be a breaking deal for us.

Also:

1. I understand that you will implement error visualizations as the next task, am I right? The current visualization above the nodes with so strong red etc is a little bit too strong and I'd prefer to release errors in the package in a nicer form. Just confirming.

2. There are anti-aliasing visual glitches here. I understand it is being worked on as a separate issue, right?

**Comment by farmaazon** [comment]
> No, we don't care about backward compatibility before public release. **However, it is important that people who were running Enso so far, after downloading the new version will have everything working**

Well, for me the highlighted sentence is an exact definition of backward compatibility :) So actually we do care about it.

> 1. I understand that you will implement error visualizations as the next task, am I right? The current visualization above the nodes with so strong red etc is a little bit too strong and I'd prefer to release errors in the package in a nicer form. Just confirming.

Yes. I shall finish next task during this sprint.

> 2. There are anti-aliasing visual glitches here. I understand it is being worked on as a separate issue, right?

Yes. https://github.com/enso-org/ide/issues/1180

**Comment by akavel** [comment]
@wdanilo Honestly, esp. given that enso is clearly still in alpha and a lot of other things are still in imperfect alpha state (just to begin with the textual editor...), I'd love if #1180 was not a blocker for merging #1182 into a release when the latter is functionally done. For me seeing errors would be such an immense boost in usability, that I'm waiting with any further experiments with enso IDE until they're available, as I was not able to develop anything without them. I'd howl in despair if the missing antialiasing (which I didn't even notice until I read about it in the comment above) was the only reason to postpone the error handling to a later alpha release. *PUH-LEEZE pretty pleah-seh!* ❤️ ❤️ ❤️ 🙏

File: src/rust/ide/lib/enso-protocol/src/language_server/types.rs [hunk 1]
```diff
@@ -113,55 +113,152 @@ pub enum Notification {
...
+    pub profiling_info : Vec<ProfilingInfo>,
+    pub from_cache     : bool,
+    pub payload        : ExpressionUpdatePayload
+}
+
+/// Profiling information on an executed expression. It is implemented as a union as additional
+/// types of information will be added in the future.
+#[derive(Clone,Debug,Deserialize,PartialEq,Serialize)]
+#[allow(missing_docs)]
+// Not sure what the future variants will be, and the copyness is not essential for this.
```
Comment (wdanilo) [inline src/rust/ide/lib/enso-protocol/src/language_server/types.rs]: I'd replace `copyness` with `deriving Copy`

File: src/rust/ide/src/ide/integration.rs [hunk 2]
```diff
@@ -9,7 +9,7 @@ use crate::prelude::*;
 use crate::controller::graph::NodeTrees;
 use crate::controller::searcher::action::MatchInfo;
 use crate::controller::searcher::Actions;
-use crate::model::execution_context::ComputedValueInfo;
+use crate::model::execution_context::{ComputedValueInfo, EvaluationError};
```
Comment (wdanilo) [inline src/rust/ide/src/ide/integration.rs]: wrong import

File: src/rust/ide/src/model/execution_context.rs [hunk 3]
```diff
@@ -20,6 +21,14 @@ use uuid::Uuid;
 
 
 
+// =================
+// === Constants ===
+// =================
+
+const DATAFLOW_ERROR_MESSAGE:&str = "Dataflow error.";
```
Comment (wdanilo) [inline src/rust/ide/src/model/execution_context.rs:33]: this should have docs telling when this error message will be used. For me its not clear now.

File: src/rust/ide/src/model/execution_context.rs [hunk 4]
```diff
@@ -339,7 +383,7 @@ mod tests {
 
     use crate::executor::test_utils::TestWithLocalPoolExecutor;
 
-    use enso_protocol::language_server::types::test::value_update_with_type;
+    use enso_protocol::language_server::types::test::{value_update_with_type, value_update_with_dataflow_error, value_update_with_dataflow_panic};
```
Comment (wdanilo) [inline src/rust/ide/src/model/execution_context.rs]: wrong import

**Review by wdanilo — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]

File: src/rust/ide/src/ide/initializer.rs [hunk 5]
```diff
@@ -115,6 +143,18 @@ impl Initializer {
         executor::global::spawn(project_manager.runner());
         Ok(Rc::new(project_manager))
     }
+
+    fn display_warning_on_unsupported_engine_version(&self, view:&ide_view::project::View, project:&model::Project) -> FallibleResult {
```
Comment (wdanilo) [inline src/rust/ide/src/ide/initializer.rs]: too long

File: src/rust/ide/src/ide/initializer.rs [hunk 6]
```diff
@@ -96,10 +122,12 @@ impl Initializer {
                 let project_manager = None;
                 let json_endpoint   = json_endpoint.clone();
                 let binary_endpoint = binary_endpoint.clone();
+                // TODO[ao]: we should think how to handle engine's versions in cloud.
```
Comment (wdanilo) [inline src/rust/ide/src/ide/initializer.rs:125]: add issue + link it here please

File: src/rust/ide/view/src/status_bar.rs [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+
+
+// =================
+// === Constants ===
+// =================
+
+/// The height of the status bar.
+pub const HEIGHT        : f32 = 40.0;
+/// Padding inside the status bar.
+pub const PADDING       : f32 = 12.0;
```
Comment (wdanilo) [inline src/rust/ide/view/src/status_bar.rs]: rm spaces

File: src/rust/ide/view/src/status_bar.rs [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+
+        model.label.remove_from_scene_layer_DEPRECATED(&scene.layers.main);
+        model.label.add_to_scene_layer_DEPRECATED(&scene.layers.breadcrumbs);
+
+        enso_frp::extend! { network
+            event_added       <- frp.add_event.map(f!((label) model.add_event(label)));
+            process_added     <- frp.add_process.map(f!((label) model.add_process(label)));
+            _process_finished <- frp.finish_process.filter_map(f!([model](id)
+                model.finish_process(*id).as_some(*id)
+            ));
```
Comment (wdanilo) [inline src/rust/ide/view/src/status_bar.rs]: rm `[model]`

File: src/rust/ide/view/src/status_bar.rs [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+
+impl display::Object for View {
+    fn display_object(&self) -> &display::object::Instance<Scene> { &self.model.display_object }
+}
+
+impl Deref for View {
+    type Target = Frp;
+
+    fn deref(&self) -> &Self::Target { &self.frp }
+}
```
Comment (wdanilo) [inline src/rust/ide/view/src/status_bar.rs]: nl

**Review by wdanilo — APPROVED** [review]

**Pull request merged**